Trippy/Utils/ParseUtils.h

```diff
@@ -9,18 +9,23 @@
 #import "Parse/Parse.h"
 #import "LocationCollection.h"
 #import "Location.h"
+#import "Itinerary.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
 @interface ParseUtils : NSObject
 + (NSArray *)getCollectionKeys;
 + (NSArray *)getLocationKeys;
-+ (NSString *)getLoggedInUserId;
++ (NSArray *)getItineraryKeys;
++ (NSString *)getLoggedInUsername;
++ (void) itineraryFromPFObj:(PFObject *)obj completion:(void (^)(Itinerary *itinerary, NSError *))completion;
 + (void)collectionFromPFObj:(PFObject *)obj completion:(void (^)(LocationCollection *collection, NSError *))completion;
 + (Location *)locationFromPFObj:(PFObject *)obj;
 + (PFObject *)newPFObjWithCollection:(LocationCollection *)collection;
 + (PFObject *)newPFObjWithLocation:(Location *)loc;
 + (PFObject *)oldPFObjFromLocation:(Location *)loc;
++ (PFObject *)oldPFObjFromCollection:(LocationCollection *)col;
```

Comment:
Still not a fan of the "old"s. Here's an idea - drop the olds entirely, and have the news check for the olds first, returning if present. Note that if you find an old, but the data doesn't match, you should probably update the old in Parse. This pattern is known as an "upsert" - update or insert.

Reply:
Actually, thinking back to this, you could keep the olds (with a change in name, granted), and instead of passing in a whole collection object, just pass the ID (since that's all you need). Also, if nothing outside of this class itself is using the method, it shouldn't be in the header.

Reply:
Gotcha...will keep in mind, although I'm hoping these methods won't be necessary at all after refactoring a bit.